framework/src/meshmodifiers/SideSetsAroundSubdomain.C

```diff
@@ -58,6 +62,9 @@ SideSetsAroundSubdomain::modify()
   if (!_mesh_ptr)
     mooseError("_mesh_ptr must be initialized before calling SideSetsAroundSubdomain::modify()!");
 
+  // Initialize the BlockRestrictable parent
+  initializeBlockRestrictable(_pars);
```

Comment:
That's not really better overall in terms of design since now an outside class has to initialize another unrelated class. It does however remove the burden from a future class designer from running into this issue. Right now this is the only MeshModifier inheriting from BlockRestrictable (believe it or not) so I didn't want to over design.

I think a better design might be to call a non-virtual on the BASE CLASS that initializes itself and then calls the existing "modify()" method. That keeps the initialization all internal.